monitoring-engine/src/main/java/io/jenkins/plugins/remotingopentelemetry/engine/config/ConfigParser.java

```diff
@@ -0,0 +1,86 @@
+package io.jenkins.plugins.remotingopentelemetry.engine.config;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Load configuration from Environment variables, config file, etc...
+ * Only environment variables are supported for now.
+ */
+public final class ConfigParser {
+    private final Object bean;
+    private Map<String, String> env = System.getenv();
+
+    public ConfigParser(Object bean) {
+        this.bean = bean;
+    }
+
+    public void setEnv(Map<String, String> env) {
+        this.env = env;
+    }
+
+    public void parse() throws ConfigurationParseException {
+        try {
+            doParse();
+        } catch (ConfigurationParseException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
It is not the best practice in general, because it would catch 'Error's that should be fatal. In this case it leads to a failure